Sources/Components/Button/Button/Button.swift

```diff
@@ -10,6 +10,7 @@ public class Button: UIButton {
     private let cornerRadius: CGFloat = 8.0
     private var titleHeight: CGFloat?
     private var titleWidth: CGFloat?
+    var userInterfaceStyle: UserInterfaceStyle?
```

Comment:
I feel like this is not necessary. I was imagining a static theme with all the defined colors. 

Something like:
```
struct Theme {
    static var current: Theme {
        // dark or light
    }

    var background: UIColor
    var foreground: UIColor
    var link: UIColor
    // ... etc
}
```

Then use it like this maybe:

```
self.backgroundColor = Theme.current.background
```

Reply:
How we would check the current user style, tho? Should we keep a global state? Grab it from the UIWindow? 

Haven't come up with a proper solution other than passing it from the view.

Reply:
I don't know how it is going to work, but I would guess we could grab it from UIWindow when we create it in AppDelegate.

Reply:
Or! We might want to start using the appearance api. Implement this method in all the components: `appearance(for trait: UITraitCollection)`

Reply:
It's tricky since the only one that knows the trait collection is the view and view controllers. The UIWindow has no knowledge of the UIUserInterfaceStyle as far as I know. 

Also, when iOS switches to dark or light mode, only UIViewControllers are informed of this change, the UIWindow or the AppDelegate don't know about it.

Reply:
Isn't UIWindow a UIView? 😄

Reply:
One question is also if we want to support that users can override dark or light in our app or if we should only follow the iOS setting?

Reply:
UIWindow is a UITraitEnvironment 😉

Reply:
> Isn't UIWindow a UIView? 😄

Haha, true! Doesn't change much since it still doesn't get informed for changes in the UITraitCollection.

> One question is also if we want to support that users can override dark or light in our app or if we should only follow the iOS setting?

I think the proper implementation is to react to changes. Only following the iOS setting could be a viable option as well if we don't mind the lack of responsiveness.

I think we need to decide what we want to do.

Reply:
I'll do an iteration without listening to changes.

Reply:
This is the alternative https://github.com/finn-no/FinniversKit/pull/517

Reply:
Whether we do live updates in the style or if we set the style when the app launches it's something that should be decided by the designer, PM, and TDE with feedback from the developers.

And we should decide soon so we can continue the progress with this task :)